src/PrestaShopBundle/DependencyInjection/Compiler/LoadModulesAsBundlesPass.php

```diff
@@ -0,0 +1,174 @@
+use Symfony\Component\Config\FileLocator;
+
+/**
+ * Load services stored in modules.
+ */
+class LoadModulesAsBundlesPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $this->registerServicesFromModules($container);
+        $this->registerTemplatesFromModules($container);
+    }
+
+    /**
+     * Load all services registered in every module.
+     *
+     * @todo: only register services for installed modules.
+     *
+     * @param ContainerBuilder $container
+     * @return void
+     */
+    private function registerServicesFromModules(ContainerBuilder $container)
+    {
+        $modulesPaths = $this->getModulesPaths();
+
+        foreach ($modulesPaths as $modulePath) {
+            if (file_exists($modulePath.'/config/services.yml')) {
+                $loader = new YamlFileLoader($container, new FileLocator($modulePath.'/config/'));
+                $loader->load('services.yml');
+            }
+        }
+    }
+
+    /**
+     * Allow the PrestaShop modules to override Twig templates
+     *
+     * @param ContainerBuilder $container
+     * @return void
+     */
+    private function registerTemplatesFromModules(ContainerBuilder $container)
+    {
+        $twigFilesystemLoaderDefinition = $container->getDefinition('twig.loader.filesystem');
+
+        $bundleHierarchy = $this->getBundleHierarchy($container);
+
+        foreach ($bundleHierarchy as $name => $bundle) {
+            $namespace = $this->normalizeBundleName($name);
+
+            foreach ($this->getModulesPaths() as $modulePath) {
+                if (is_dir($dir = $modulePath.'/views/'.$namespace)) {
+                    $twigFilesystemLoaderDefinition->addMethodCall('addPath', array($dir, $namespace));
+                    /* Template from modules MUST be the first called if any */
+                    $twigFilesystemLoaderDefinition->setMethodCalls(array_reverse($twigFilesystemLoaderDefinition->getMethodCalls()));
+
+                    $container->addResource(new FileExistenceResource($dir));
+                }
+            }
+        }
+    }
+
+    /**
+     * @return \Iterator
+     */
+    private function getModulesPaths()
+    {
+        return Finder::create()->directories()->in(__DIR__.'/../../../../modules')->depth(0);
+    }
+
+    /**
+     * Return the Bundle hierarchy structure.
+     *
+     * @param ContainerBuilder $container
+     * @return array
+     */
+    private function getBundleHierarchy(ContainerBuilder $container)
+    {
+        $bundleHierarchy = array();
+
+        foreach ($container->getParameter('kernel.bundles_metadata') as $name => $bundle) {
+            if (!array_key_exists($name, $bundleHierarchy)) {
+                $bundleHierarchy[$name] = array(
+                    'paths' => array(),
+                    'parents' => array(),
+                    'children' => array(),
+                );
```

Comment:
Why bother creating this structure if all you use afterwards are the bundle names?

Reply:
Already fixed, sorry for noise. I add the WIP tag btw :)